ci/build_docs.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create test conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file_key docs \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee requirements.txt
+
+rapids-mamba-retry create -n test
+conda activate test
+pip install -r requirements.txt
+
+rapids-print-env
+
+rapids-logger "Downloading artifacts from previous jobs"
+VERSION_NUMBER=$(rapids-get-rapids-version-from-git)
+
+# Build CPP docs
+gpuci_logger "Build CPP docs"
+pushd docs
+sphinx-build -b dirhtml . _build
```

Comment:
```suggestion
sphinx-build -b dirhtml . _html -W
sphinx-build -b text . _text -W
```

AjayThorve added this flag in https://github.com/rapidsai/cuml/pull/5155, which I think is a good idea to add to all of the repositories.

It will treat warnings as errors as indicated by the sphinx docs: https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W

If any errors appear as a result of this, we can try to resolve them ourselves or solicit feedback from the `rapids-cmake` team.